src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,61 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type.equals(Truck.class)) {
```

Comment:
When comparing Class objects in Java, it's more typical to use == rather than equals. This is because there's only one Class object for each type, so you can safely use identity comparison (==).